File: app/src/main/java/com/miguelgaeta/AppActivity.java
package com.miguelgaeta;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.miguelgaeta.backgrounded.Backgrounded;

import rx.functions.Action1;

public class AppActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Backgrounded.init(getApplication());
        Backgrounded.get().subscribe(new Action1<Boolean>() {

            @Override
            public void call(Boolean backgrounded) {

                Log.e("Test", "Backgrounded status: " + backgrounded);
            }
        });
    }
}

File: app/src/main/java/com/miguelgaeta/AppActivity.kt
package com.miguelgaeta

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.miguelgaeta.backgrounded.Backgrounded

class AppActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        Backgrounded.init(application)
        Backgrounded.get().subscribe { backgrounded -> Log.e("Test", "Backgrounded status: " + backgrounded) }

        Log.e("Test", "Currently backgrounded: " + Backgrounded.isBackgrounded())
    }

    override fun onDestroy() {
        super.onDestroy()

        Backgrounded.destroy(application)
    }
}
